R/descriptives.R

```diff
@@ -1881,3 +1948,339 @@ Descriptives <- function(jaspResults, dataset, options) {
 
   return(data)
 }
+
+.descriptivesLikertPlots <- function(dataset, options, name) {
+
+  leng <- length(dataset)
+  depends <- c("descriptivesLikertPlot", "splitby", "variables", "fontSizeLikert")
+
+  likPlot <- createJaspPlot(title = name, dependencies = depends, width = 1300, height = if (leng == 1) 250 else 200*(leng*0.8))
+  errorMessage <- .descriptivesCheckPlotErrors(dataset, names(dataset), obsAmount = "< 2")
+  if (!is.null(errorMessage)) {
+    likPlot$setError(gettextf("Plotting not possible: %s", errorMessage))
+    return(likPlot)
+  }
+
+  # Likert Part: Preparing & summarize data in the likert format (% of levels per variable)
+  nLevels <- nlevels(factor(dataset[, 1]))
+  center <- (nLevels - 1)/2 + 1
+  lowRange <- 1:floor(center - 0.5)
+  highRange <- ceiling(center + 0.5):nLevels
+
+  if (!all(sapply(dataset, function(x) nlevels(factor(x))) == nLevels)) {
+    likPlot$setError(gettext("All items (columns) must have the same number of levels!"))
+    return(likPlot)
+  }
+  if (center < 1.5) {
+    likPlot$setError(gettext("Items must have 2 or more levels!"))
+    return(likPlot)
+  }
+
+  # Summarizing item contribution and get clean data
+  results <- data.frame()
+  results <- data.frame(Response = 1:nLevels)
+  for (i in 1:ncol(dataset)) {
```

Comment:
```suggestion
  results <- data.frame(Response = seq_len(nLevels))
  for (i in seq_len(ncol(dataset))) {
```
in general, always use `seq_len` and `seq_along` instead of `1:n` and `1:length(x)`.